### PR TITLE
Refactored to handle error and test for wrong http request

### PR DESCRIPTION
**Pull request by jshamoo**
Hi team, I refactored my code to handle errors so it doesn't break my endpoints and added a test for 404 response. Please review. Thanks.
cc: @jonyang221 @rlnolan15 @Evillivekai @j-gens

### CODE REVIEW

File: client/components/app.jsx [hunk 1]
```diff
@@ -23,12 +24,14 @@ class App extends React.Component {
     $.ajax({
       url: '/api/movie',
       method: 'GET',
-      data: {id: 4},
+      data: {id: 3},
       success: (res) => {
         console.log('clent got data:', res);
```
Comment (j-gens) [inline client/components/app.jsx]: a small spelling error - 'client'

File: client/components/app.jsx [hunk 2]
```diff
@@ -38,6 +41,14 @@ class App extends React.Component {
 
 
   render() {
+    if (this.state.error) {
+      return (
+          <ErrorMessage>
+            Oops, this movie doesn't exist.
+        </ErrorMessage>
+      )
+    }
```
Comment (j-gens) [inline client/components/app.jsx]: this is great! :)

File: server/app.js [hunk 3]
```diff
@@ -12,12 +12,22 @@ app.use(express.static('public'));
...
-    casts.sort((a, b) => {
-      return a.role - b.role;
+  db.getCasts(id)
+    .then(data => {
+      if (data.length === 0) {
+        res.sendStatus(404);
+      } else {
+
+        data[0].casts.sort((a, b) => {
+          return a.role - b.role;
```
Comment (therobinkim) [inline server/app.js]: Are we alphabetizing here? Or something else?

File: server/app.js [hunk 4]
```diff
@@ -12,12 +12,22 @@ app.use(express.static('public'));
...
+        });
+        res.send(data[0].casts);
```
Comment (therobinkim) [inline server/app.js]: `res.json` might be a better choice

File: server/app.js [hunk 5]
```diff
@@ -12,12 +12,22 @@ app.use(express.static('public'));
...
+      }
+    })
+    .catch(err => {
+      console.log(err);
```
Comment (therobinkim) [inline server/app.js]: We should always end the response

**Pull request merged**